lib/pf/Switch/Xirrus/AP_http.pm

```diff
@@ -84,27 +84,31 @@ Overriding the default implementation for the external captive portal
 =cut
 
 sub returnRadiusAccessAccept {
-    my ($self, $vlan, $mac, $port, $connection_type, $user_name, $ssid, $wasInline, $user_role) = @_;
+    my ($self, $args) = @_;
     my $logger = $self->logger;
 
+    my $filter = pf::access_filter::radius->new;
+    my $rule = $filter->test('returnRadiusAccessAccept', $args);
     my $radius_reply_ref = {};
 
-    my $node = node_view($mac);
+    my $node = $args->{'node_info'};
 
-    my $violation = pf::violation::violation_view_top($mac);
-    # if user is unregistered or is in violation then we reject him to show him the captive portal
+    my $violation = pf::violation::violation_view_top($args->{'mac'});
+    # if user is unregistered or is in violation then we reject him to show him the captive portal 
     if ( $node->{status} eq $pf::node::STATUS_UNREGISTERED || defined($violation) ){
-        $logger->info("[$mac] is unregistered. Refusing access to force the eCWP");
+        $logger->info("[$args->{'mac'}] is unregistered. Refusing access to force the eCWP");
         my $radius_reply_ref = {
             'Tunnel-Medium-Type' => $RADIUS::ETHERNET,
             'Tunnel-Type' => $RADIUS::VLAN,
             'Tunnel-Private-Group-ID' => -1,
         };
+        $radius_reply_ref = $filter->handleAnswerInRule($rule,$args,$radius_reply_ref);
         return [$RADIUS::RLM_MODULE_OK, %$radius_reply_ref];
 
     }
     else{
         $logger->info("Returning ACCEPT");
+        $radius_reply_ref = $filter->handleAnswerInRule($rule,$args,$radius_reply_ref);
```

Comment:
once again :)

That's why a subroutine would be ideal, so much duplication

![image](https://cloud.githubusercontent.com/assets/3857942/10888357/bfd2bea6-8159-11e5-991d-ff5c2694f098.png)